src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1312,6 +1312,8 @@ protected void ResetEngineInternal()
             EngineController.MessageLogged += LogMessage;
             EngineController.TraceReconcliationComplete += EngineController_TraceReconcliationComplete;
 
+            CustomNodeManager.EngineController = EngineController;
```

Comment:
Why would you prefer setting it directly instead of also passing it?

Reply:
I don't prefer. It was easier for me.

Reply:
And there are some cases that EngineController is redefined, so we need to update the EngineController inside CustomNodeManager.